pyiron_base/jobs/job/executor.py

```diff
@@ -0,0 +1,64 @@
+from pyiron_base.jobs.job.generic import GenericJob
+
+
+class JobWithExecutor(GenericJob):
+    def __init__(self, project, job_name):
+        super().__init__(project, job_name)
+        self._executor_type = None
+
+    @property
+    def executor_type(self):
+        return self._executor_type
+
+    @executor_type.setter
+    def executor_type(self, exe):
+        self._executor_type = exe
```

Comment:
At a minimum, the setter should be used as an opportunity to fail early by comparing to a set of allowed values and raising an exception if it's not there.

At best, we can actually relax the allowable executors by just making sure it's available and that it's likely to follow the right UI (could use `inspect` to really verify the signature, but I won't do that here). 

Add the following imports:

```python
from concurrent.futures import Executor as StdLibExecutor
from inspect import isclass

from h5io import _import_class 
```
(Or redefine it somewhere more convenient is probably better, I just want to minimize my typing in the github review browser interface)

Then

```suggestion
    @staticmethod
    def _is_a_subclass_of_executor(obj):
        return isclass(exe) and issubclass(exe, StdLibExecutor):

    @executor_type.setter
    def executor_type(self, exe):
        if isinstance(exe, str):
            try:
                exe_class = _import_class(exe)  # Make sure it's available
                if not self._is_a_subclass_of_executor(exe_class):  # And what we want
                    raise TypeError(
                        f"{exe} imported OK, but {exe_class} is not a subclass of {StdLibExecutor}"
                    )
            except Exception as e:
                raise ImportError("Something went wrong trying to import {exe}") from e
        elif isclass(exe) and issubclass(exe, StdLibExecutor):
            self._executor_type = f"{exe.__module__}.{exe.__name__}"
        elif isinstance(exe, StdLibExecutor):
            raise NotImplementedError(
                "We don't want to let you pass an entire executor, because you might think its state comes "
                "with it. Try passing `.__class__` on this object instead."
            )
        else:
            raise TypeError(
                f"Expected an executor class or string representing one, but got {exe}"
            )
```

Reply:
Note that this is also a bit of a UI change, since we insist on `"concurrent.futures."` even ahead of the standard thread and process pool guys. If you still want to accept just the class name for these two special and common cases, it's easy enough to add the extra logic here.

Reply:
I currently tend to prefer white listing selected executors, rather than supporting all of them by default. This allows us to test the executors to check if they are working with `pyiron_base`. For example adding support for `dask` is technically possible, but I would prefer somebody test it first before we add it to the white list.

Reply:
>but I would prefer somebody test it first before we add it to the white list.

Per my main comment in the re-review, by having a strict whitelist we force that person to clone and modify `pyiron_base` to do this test. A more relaxed approach lets that testing happen without modifications to base itself.